### PR TITLE
fix: add publicUrl to exclude promisifyAll

### LINKED ISSUE #1344: file.publicUrl() returns a Promise instead of the publicUrl
#### Steps to reproduce
    const bucket = storage.bucket(bucketName);
    const file = bucket.file(filename);
    const publicUrl = file.publicUrl();

### PR DESCRIPTION
**Pull request by fernandolguevara**
Fixes #1344 

I exclude "publicUrl" from the list of functions that will use promises

### CODE REVIEW

File: src/file.ts [hunk 1]
```diff
@@ -3798,7 +3798,7 @@ class File extends ServiceObject<File> {
  * that a callback is omitted.
  */
 promisifyAll(File, {
-  exclude: ['request', 'setEncryptionKey'],
+  exclude: ['request', 'setEncryptionKey', 'publicUrl'],
```
Comment (AVaksman) [inline src/file.ts:3801]: Please place `'publicUrl'` in alphabetic order.

Also please add `'publicUrl'` to the list to fix unit tests
https://github.com/googleapis/nodejs-storage/blob/642a9dff611be18b8b8ec6a14cfaed873daaf9f4/test/file.ts#L77

**Comment by AVaksman** [comment]
@fernandolguevara Thank you for contribution.

Just a couple of nits.

**Review by AVaksman — DISMISSED** [review]

File: test/file.ts [hunk 2]
```diff
@@ -74,7 +74,7 @@ const fakePromisify = {
     }
 
     promisified = true;
-    assert.deepStrictEqual(options.exclude, ['request', 'setEncryptionKey']);
+    assert.deepStrictEqual(options.exclude, ['publicUrl', 'request', 'setEncryptionKey']);
```
Comment (shaffeeullah) [inline test/file.ts:77]:
```suggestion
    assert.deepStrictEqual(options.exclude, [
      'publicUrl',
      'request',
      'setEncryptionKey',
    ]);
```

**Comment by shaffeeullah** [comment]
@fernandolguevara friendly bump. If you do not have time to implement the requested changes, let me know and I can take it over from here. However, please sign the CLA so this PR can be merged. Thanks for making these changes!

**Comment by fernandolguevara** [comment]
@googlebot I signed it!

**Review by shaffeeullah — APPROVED** [review]

**Review by stephenplusplus — APPROVED** [review]

**Pull request merged**